### ISSUE TITLE
Characters in canvas are cropped

### ISSUE THREAD

**Issue by notoriousmango** [mozilla/pdf.js#12763]
Attach (recommended) or Link to PDF file here:
[pdfjsError.pdf](https://github.com/mozilla/pdf.js/files/5722039/pdfjsError.pdf)

Configuration:
I tested using https://mozilla.github.io/pdf.js/web/viewer.html
- Web browser and its version: Chrome 87.0.4280.88
- Operating system and its version: Mac OS 11.0.1
- PDF.js version: https://mozilla.github.io/pdf.js/web/viewer.html
- Is a browser extension: No

Steps to reproduce the problem:
1. go to https://mozilla.github.io/pdf.js/web/viewer.html
2. open the pdf above

What is the expected behavior? (add screenshot)
![image](https://user-images.githubusercontent.com/32555977/102733996-0a25f000-4382-11eb-8b5e-d9e3f255f989.png)
Characters in bubbles are not cropped
What went wrong? (add screenshot)
![image](https://user-images.githubusercontent.com/32555977/102734019-1d38c000-4382-11eb-9167-0efabe25cec7.png)
Characters in bubbles are cropped

**Comment by Snuffleupagus** [comment]
WFM, with PDF.js version `2.7.345` in Firefox Nightly 86 on Windows.

> Web browser and its version: Chrome 87.0.4280.88
> Operating system and its version: Mac OS 11.0.1

This does look somewhat like platform and/or browser-specific issue, rather than a PDF.js one, since otherwise you'd sort of expect it to be consistently broken *everywhere*.

**Comment by tamuratak** [comment]
I can reproduce the issue with Firefox 84.0 on https://mozilla.github.io/pdf.js/web/viewer.html.

macOS Mojave 10.14.6
Firefox 84.0
PDF.js 2.7.421

![2020-12-21 20 36 09](https://user-images.githubusercontent.com/10665499/102773292-58f87780-43cc-11eb-8e7b-6be24a9c9d78.png)

Interestingly, when I change the zoom level to 50%, there is almost no cropping.

![2020-12-21 20 43 40](https://user-images.githubusercontent.com/10665499/102773974-6feb9980-43cd-11eb-836b-197af1ab2361.png)

**Comment by Snuffleupagus** [comment]
Is this still reproducible, in the latest version of Firefox on MacOS?

**Comment by calixteman** [comment]
Yes I can reproduce with Firefox nightly on MacOS (Ventura 13.6).

**Comment by calixteman** [comment]
Here's a reduced test case:
[issue12763_reduced.pdf](https://github.com/mozilla/pdf.js/files/12708614/issue12763_reduced.pdf)

**Comment by calixteman** [comment]
In using https://mozilla.github.io/pdf.js/web/viewer.html, I can reproduce on mac with Chrome but not with Safari.
@jfkthame, do you have an idea on what could be wrong here ?
